roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
@fullyint, I'm able to replicate your issue now. Thanks again. This is a bit of a bummer. I don't see any clear way forward except (1) trying to make intelligent guesses based on related connection variables or (2) requiring `ansible_connection` to be set via inventory or extra vars. I suppose the latter is best and can be documented.